### PR TITLE
Change db_server Docker image to be based on Google's MySQL 5.7 image

### PR DESCRIPTION
**Pull request by RJPercival**
This base image is already approved for distribution by Google, which makes it easier to get approval for distributing our image.

### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: integration/docker_compose_integration_test.sh [hunk 1]
```diff
@@ -6,6 +6,10 @@ readonly DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
 docker_compose_up() {
   local http_addr="$1"
 
+  # Configure Docker for Google Cloud authentication, so that images stored in
+  # Google Container Repository are accessible.
+  gcloud auth configure-docker
```
Comment (daviddrysdale) [inline integration/docker_compose_integration_test.sh]: So this was the magic `--work-properly` flag that was missing from #1402 ?

Comment (RJPercival) [reply integration/docker_compose_integration_test.sh]: I believe so. It appears that we have a fairly old version of the gcloud SDK cached in Travis however, which doesn't have this command. I've added a commit that'll hopefully fix that.

Comment (RJPercival) [reply integration/docker_compose_integration_test.sh]: It turned out that actually, while we added a "google-cloud-sdk" directory to Travis' cache, we never actually installed the gcloud SDK into that directory. 

https://github.com/google/trillian/blob/991b3a6de176770c5562a8e09afd94ae6e19df15/.travis.yml#L14-L16

What we were using was the version of gcloud that's provided by Travis, which is a few years old apparently. I'm going to get Travis to download and install the latest version of gcloud.

Comment (RJPercival) [reply integration/docker_compose_integration_test.sh]: That's made it possible to use `gcloud auth configure-docker`, but unfortunately I'm still seeing the same error message on Travis:
```
ERROR: Service 'mysql' failed to build: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication
```

Comment (RJPercival) [reply integration/docker_compose_integration_test.sh]: I've ended up mirroring the mysql5 image into our container registry, which gets around the authentication issues. I've configured Google Cloud Build to update our mirror of that image every time a commit is pushed to master, which will keep it reasonably fresh.

The alternative would be to only build the Docker images (and therefore only run the docker-compose integration test) when master is updated (i.e. not when testing PRs). In that case, Travis could login using Google Cloud service account credentials in order to access the original mysql5 image. That has the downside of only finding out about breakages after a PR is merged and having to give Travis service account credentials, but avoids us having to mirror the image.

**Pull request merged**